Tests/RegexBuilderTests/RegexDSLTests.swift

```diff
@@ -1071,6 +1071,63 @@ class RegexDSLTests: XCTestCase {
       }
     }
   }
+
+  func testTypeErasedRegexInDSL() throws {
+    do {
+      let input = "johnappleseed: 12."
+      let numberRegex = try! Regex(#"(\d+)\.?"#)
```

Comment:
Added one.  The idea is to fix up the search bounds so that `^` refers to the start of the input.

It happens to work today because anchors currently use the base string's bounds, not the input slice's bounds.

Reply:
Right, there is a bug here that another bug is masking, but when that other bug is fixed, how do we fix this bug?

Reply:
You can illustrate the difference with a substring input, where the subject bounds are the substring's bounds and the search bounds are contained within.